File: 백준/Gold/2225. 합분해/합분해.py
import sys

def solution():
  N, K = map(int, sys.stdin.readline().split())
  dp = [[1 for _ in range(K + 1)] for _ in range(N + 1)]
  dp[0][0] = 0
  for i in range(1, N + 1):
    for j in range(1, K + 1):
      if i == 1:
        dp[i][j] = j
      elif j == 1:
        dp[i][j] = 1
      else:
        dp[i][j] = dp[i - 1][j] + dp[i][j - 1]
  print(dp[N][K] % 1000000000)


solution()

File: 백준/Gold/2225. 합분해/합분해.js
let fs = require('fs');
let inputs = fs.readFileSync('/dev/stdin').toString().split(' ');
const [N, K] = inputs.map(Number);

const dp = Array.from({length: N + 1}, () => Array.from({length: K + 1}, () => BigInt(0)));

for (let j = 0; j <= K; j++) {
    dp[0][j] = BigInt(1);
}

for (let i = 0; i <= N; i++) {
    dp[i][1] = BigInt(1);
}


for (let i = 1; i <= N; i++) {
    for (let j = 1; j <= K; j++) {
        dp[i][j] = (dp[i - 1][j] + dp[i][j - 1]) % BigInt(1000000000);
    }
}

console.log(dp[N][K].toString());